inkfish/proof_of_time.py

```diff
@@ -31,38 +32,104 @@ def deserialize_proof(proof_blob,  discriminant):
     return [ClassGroup.from_bytes(blob, discriminant) for blob in proof_arr]
 
 
-def create_proof_of_time_large(discriminant, generator, iterations, int_size_bits):
+def create_proof_of_time_wesolowski(discriminant, x, iterations, int_size_bits):
+    L, k, _ = proof_wesolowski.approximate_parameters(iterations)
+
+    powers_to_calculate = [i * k * L for i in range(0, math.ceil(iterations/(k*L)) + 1)]
+    powers_to_calculate += [iterations]
+    powers = iterate_squarings(x, powers_to_calculate)
+
+    y = powers[iterations]
+    identity = ClassGroup.identity_for_discriminant(discriminant)
+    proof = proof_wesolowski.generate_proof(identity, x, y, iterations, k, L, powers)
+    return y.serialize(), serialize_proof([proof])
+
+
+def create_proof_of_time_nwesolowski(discriminant, x, iterations,
+                                     int_size_bits, depth_limit, depth=0):
+    """
+    Returns a serialized proof blob, using n_wesolowski
+                     iterations_1                        iterations_2      proof_2
+     [----------------------------------------------|---------------------][-----]
+                                                    |---------------------]
+                                                            proof_1
+    """
+    L, k, w = proof_wesolowski.approximate_parameters(iterations)
+
+    iterations_1 = (iterations * w) // (w + 1)
+    iterations_2 = iterations - iterations_1
+
+    identity = ClassGroup.identity_for_discriminant(discriminant)
+
+    powers_to_calculate = [i * k * L for i in range(0, math.ceil(iterations_1/(k*L)) + 1)]
+    powers_to_calculate += [iterations_1]
+
+    powers = iterate_squarings(x, powers_to_calculate)
+    y_1 = powers[iterations_1]
+
+    receive_con, send_con = Pipe(False)
+    p = Process(target=proof_wesolowski.generate_proof,
```

Comment:
This is great you got this working, and it's not even too nasty-looking. You might optionally consider using `multiprocessing.Pool` for an even simpler and cleaner version. https://docs.python.org/2/library/multiprocessing.html#multiprocessing.pool.multiprocessing.Pool.apply_async